### PR TITLE
Import DBGOVL32 into this repository.

### PR DESCRIPTION
**Pull request by statham-arm**

### CODE REVIEW

File: dbgovl32/dbgovl32.rst [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+
+-  Each program segment described by a program header PH of type
+   ``PT_LOAD`` shall occupy a different extent [PH.p\_offset, PH.p\_offset
+   + PH.p\_filesz) in the ELF file. (An empty extent shall not overlap
+   any other extent).
+
+In addition, a linker claiming to support debugging of overlaid programs
+shall do *at least one* of the following.
+
+-  | Provide a means to retain all original relocations in the
```
Comment (stuij) [inline dbgovl32/dbgovl32.rst:607]: this is purely stylistic, but for some reason 'provide..' is placed half a line below the bullet point

File: dbgovl32/dbgovl32.rst [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+required to support this extension but tools that support debugging
+overlaid programs should do so in one of the ways specified in
+`The ABI Extension`_.
+
+Keywords
+--------
+
+Debugging ABI for the Arm Architecture; debugging; ABI
+
+Preamble
```
[the remark this answers is not in the document]
Comment (statham-arm) [reply dbgovl32/dbgovl32.rst:61]: Oops! Must have been a copy-paste goof.

File: dbgovl32/dbgovl32.rst (hunk 1 above)
Comment (statham-arm) [reply dbgovl32/dbgovl32.rst:607]: That seems to be a rst2pdf formatting bug that occurs when you use the prefix `|` to arrange a nonstandard paragraph with known line breaks. Probably the easiest fix is to just stop trying to get that line break in there; it doesn't seem semantic.

**Comment by stuij** [comment]
There was a merge conflict, and after resolving through the web interface, we still couldn't merge and introduced an ugly merge commit. I ended up pushing a clean version to master, so this pull request is merged.

**Pull request closed without merging**